### PR TITLE
Add Drupal 8 integration

### PR DESCRIPTION
**Pull request by seemk**
### Description

Drupal Integration split from https://github.com/signalfx/signalfx-php-tracing/pull/26

A cut down (although still large) version of Drupal is added to `tests/Frameworks` just as is done with other frameworks, e.g. WordPress. This inflates the PR size, but the relevant changes can be seen at: https://github.com/signalfx/signalfx-php-tracing/commit/9cbf4b9adb19a8ff1ab47315a648461a0c51ef84

### Readiness checklist
- [x] (only for Members) Changelog has been added to the release document.
- [x] Tests added for this feature/bug.

### CODE REVIEW

File: src/DDTrace/Integrations/Drupal/DrupalIntegration.php [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        );
+
+        \DDTrace\trace_method(
+            'Drupal\Core\DrupalKernel',
+            'handleException',
+            function (SpanData $span, $args, $retval) use ($rootSpan) {
+                $span->name = 'drupal.kernel.handleException';
+                $span->meta[Tag::COMPONENT] = 'drupal';
+                if (!(isset($retval) && \method_exists($retval, 'getStatusCode') && $retval->getStatusCode() < 500)) {
+                    $rootSpan->setError($args[0]);
```
Comment (rmfitzpatrick) [inline src/DDTrace/Integrations/Drupal/DrupalIntegration.php:75]: Is the exception content captured elsewhere?

File: src/DDTrace/Integrations/Drupal/DrupalIntegration.php [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+            'handle',
+            function (SpanData $span, $args) use ($rootSpan) {
+                if (!isset($args[0])) {
+                    return;
+                }
+
+                $req = $args[0];
+
+                $route = $req->getPathInfo();
+                $rootSpan->overwriteOperationName($route);
```
Comment (rmfitzpatrick) [inline src/DDTrace/Integrations/Drupal/DrupalIntegration.php:60]: Could these be ~uuids (raw paths) or are we able to obtain route/controller info here?  `http.url` is preferable to operation names for cardinality concerns and ~`drupal.handle` may be safer.  Also, where should be we setting `http.url`, `http.method`, `http.status_code`, etc?

File: src/DDTrace/Integrations/Drupal/DrupalIntegration.php (hunk 1 above)
Comment (seemk) [reply src/DDTrace/Integrations/Drupal/DrupalIntegration.php]: Removed the if here, my initial thinking was that the exception is set at `Bootstrap.php`, but it only sets the error message, not the exception.

So `setError` is always called when handling `drupal.kernel.handleException`

File: src/DDTrace/Integrations/Drupal/DrupalIntegration.php (hunk 2 above)
Comment (seemk) [reply src/DDTrace/Integrations/Drupal/DrupalIntegration.php]: Updated - this uses the Drupal route where possible, e.g. previously the route was `/user/login`, with a Drupal route it is `user.login`. However the route information is not available all the time, so the operation name comes from Symfony instrumentation - `symfony.request`. Not ideal, but gets rid of the cardinality problem.

**Review by owais — APPROVED** [review]

**Comment by iNikem** [comment]
@seemk @owais @rmfitzpatrick can this be merged now?

**Comment by seemk** [comment]
^ Added a simple heuristic that is enabled by default to get the endpoint cardinality down. Can be turned off by setting `SIGNALFX_DRUPAL_RENAME_ROOT_SPAN` to false.

1. Language codes in the first part of the url are turned into `{lang}` wildcard
2. URLs with >= 3 parts have the last section of the route replaced with `?`
3. Route normalizer is used (numbers, uuids) are replaced with `?`

**Comment by seemk** [comment]
@rmfitzpatrick @owais If there are no additional comments regarding the route rewriting, I will merge it

**Pull request merged**